File: Data Structures & Algorithms/products-of-array-discluding-self/submission-7.py
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        #[1,2,4,6]
        #[48, 24, 12, 8]

        n = len(nums)

        result = [0]*n
        product = 1

        for i in range(n):
            j = 0

            while j < n:
                if i == j:
                    j+=1
                
                if j == n:
                    break
                
                product *= nums[j]
                j+=1
                
            result[i] = product
            product = 1

        return result

File: Data Structures & Algorithms/top-k-elements-in-list/submission-7.js
class Solution {
    /**
     * @param {number[]} nums
     * @param {number} k
     * @return {number[]}
     */
    topKFrequent(nums, k) {
        const f_map = new Map();

        for(var i = 0; i < nums.length; i++){
            f_map.set(nums[i], (f_map.get(nums[i]) ?? 0) + 1);
        }

        const entries = Array.from(f_map.entries());

        entries.sort((a,b) => b[1] - a[1]);

        const result = [];

        for(let i =0; i<k; i++){
            result.push(entries[i][0]);
        }

        return result;


        

    }
}
